### PR TITLE
Add parent class

### PR DESCRIPTION
**Pull request by meganaslattery**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPB-922

### Change description ###

Refactor CaseData.java class to add more fields

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -171,7 +171,7 @@ def versions = [
         commonsIO                       : '2.6',
         commonsValidator                : '1.6',
         restAssured                     : '3.0.7',
-        lombok                          : '1.18.12',
+        lombok                          : '1.18.14',
```
Comment (jackmunkonge) [inline build.gradle]: Check the lombok changelog, it says not to use this version as it's broken and to use 1.18.16

**Review by jackmunkonge — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (meganaslattery) [reply build.gradle]: Ok nice

Comment (meganaslattery) [reply build.gradle]: Had to upgrade for the @Jacksonized annotation

**Review by jackmunkonge — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java [hunk 2]
```diff
@@ -259,10 +264,13 @@
 
     private final DocumentLink statementOfTruthDocument;
 
+    @Builder.Default
     private final List<CollectionMember<Document>> probateDocumentsGenerated = new ArrayList<>();
 
+    @Builder.Default
     private final List<CollectionMember<Document>> probateNotificationsGenerated = new ArrayList<>();
```
Comment (meganaslattery) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java:271]: #

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseDataParent.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.extern.jackson.Jacksonized;
+
+@Jacksonized
+@SuperBuilder
+@Data
+public class CaseDataParent {
+
+    protected String registrySequenceNumber;
+
+}
```
Comment (vinceholt) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseDataParent.java:14]: COULD: could have a tri class hieararchy, rename parent to CaseDataBase and fill that with all existing fields. Then  add a class that extends that, possibly called AdditionalCaseDataFields which would include new field, then have a class that extends that called CaseData with no fields

**Review by vinceholt — APPROVED** [review]

**Review by KleoG — APPROVED** [review]

**Pull request merged**